### PR TITLE
Feature/simplify sharefile to disk

### PR DESCRIPTION
**Pull request by jayckaiser**
# Feature: Simplify ShareFile_to_Disk

## Description & motivation
This is a minor refactor and cleanup of `sharefile.sharefile_to_disk` that makes the following changes:

- Add type hints and comments
- Make `ds_nodash` and `ts_nodash` arguments optional (since these can be predefined in `local_path`)
- Remove one of the two for-loops to iterate directly over the file objects on ShareFile
- Remove initialization of file feature dictionary, and use the attributes as variables directly
- Simplify the regex search to remove double-nesting.

This change came about while investigating the code needed to run the `EarthbeamDAG` in assessment contexts.

## PR Merge Priority:

- [x] Low
- [ ] Medium
- [ ] High

## Tests and QC done:
This is currently untested (hence a draft).

### CODE REVIEW

**Review by rlittle08 — COMMENTED** [review]
looks good but definitely think we should test a couple edge cases as part of larger testing during these next couple weeks

File: ea_airflow_util/callables/sharefile.py [hunk 1]
```diff
@@ -55,53 +64,45 @@ def sharefile_to_disk(sharefile_conn_id, sharefile_path, local_path, ds_nodash,
...
-    for file in files:
+    for res in remote_files:
+
+        file_name = res['FileName']
+        parent_id = res['ParentID']
+        file_path = res['ParentSemanticPath']
+        file_path_no_base = res['ParentSemanticPath'].replace(sharefile_path, '')
+        item_id = res['ItemID']
+        # size = res['Size']  # Not used for anything downstream
+        # hash = res['MD5']
```
Comment (johncmerfeld) [inline ea_airflow_util/callables/sharefile.py:79]: IMO it'd be best to just remove these

**Review by johncmerfeld — COMMENTED** [review]
Changes look good to me and as soon as we have a plan in place I'm happy to do the testing and ✅

**Review by johncmerfeld — APPROVED** [review]
@rlittle08 and I successfully tested these changes, including the delete functionality we'd been discussing. We say it's good to merge!

File: ea_airflow_util/callables/sharefile.py [hunk 2]
```diff
@@ -55,53 +64,46 @@ def sharefile_to_disk(sharefile_conn_id, sharefile_path, local_path, ds_nodash,
...
         os.makedirs(os.path.dirname(full_local_path), exist_ok=True)
 
         # download the file and hash it
         try:
-            sf_hook.download_to_disk(item_id=file['item_id'], local_path=full_local_path)
+            sf_hook.download_to_disk(item_id=item_id, local_path=full_local_path)
 
             if delete_remote:
-                sf_hook.delete(file['item_id'])
+                logging.info(f'delete_remote set to True; removing {remote_file} from ShareFile...')
```
Comment (johncmerfeld) [inline ea_airflow_util/callables/sharefile.py:105]: @jayckaiser  FYI added this for auditability
